src/pudl/transform/ferc1.py

```diff
@@ -761,56 +762,27 @@ def reconcile_table_calculations(
 
     Args:
         df: processed table.
-        tbl_meta: processed table xbrl metadata.
+        calculation_components: processed calculation component metadata.
         xbrl_factoid_name: column name of the XBRL factoid in the processed table.
         table_name: name of the PUDL table.
         params: :class:`ReconcileTableCalculations` parameters.
     """
     # If we don't have this value, we aren't doing any calculation checking:
     if params.column_to_check is None:
         return df
-
-    # skip the calculations that have any components from other tables.
-    # this could be removed/moved to when we deal with inter-table calcs.
-    inter_table_calculated_values = list(
-        tbl_meta[~tbl_meta.intra_table_calc_flag].xbrl_factoid.unique()
-    )
-    if inter_table_calculated_values:
-        logger.warning(
-            "Skipping calculated values because they are inter-table calculations: "
-            f"{inter_table_calculated_values}"
-        )
-    intra_tbl_calcs = tbl_meta[
-        tbl_meta.intra_table_calc_flag & (tbl_meta.row_type_xbrl == "calculated_value")
+    intra_tbl_calcs = calculation_components[
+        calculation_components.intra_table_calc_flag
+        & calculation_components.xbrl_factoid_calc.notnull()
     ]
     pks = pudl.metadata.classes.Resource.from_id(table_name).schema.primary_key
-    pks_wo_factoid = [col for col in pks if col != xbrl_factoid_name]
-    calculated_dfs = []
-    for calculated_factoid, calculation in zip(
-        intra_tbl_calcs.xbrl_factoid, intra_tbl_calcs.calculations
-    ):
-        calc_df = (
-            pd.merge(
-                df,
-                pd.DataFrame(json.loads(calculation)),
-                left_on=xbrl_factoid_name,
-                right_on="name",
-            )
-            # apply the weight from the calc to convey the sign before summing.
-            .assign(calculated_amount=lambda x: x[params.column_to_check] * x.weight)
-            .groupby(pks_wo_factoid, as_index=False)["calculated_amount"]
-            .sum(min_count=1)
-            .assign(**{xbrl_factoid_name: calculated_factoid})
-        )
-        calculated_dfs.append(calc_df)
-
-    calculated_df = pd.merge(
-        df, pd.concat(calculated_dfs), on=pks, how="left", validate="1:1"
-    )
-    # Force column_to_check to be a float to prevent any hijinks with calculating differences.
-    calculated_df[params.column_to_check] = calculated_df[
-        params.column_to_check
-    ].astype(float)
+    calculated_df = calculate_values_from_components(
+        data=df.rename(columns={xbrl_factoid_name: "xbrl_factoid"}),
+        calculation_components=intra_tbl_calcs,
+        validate="one_to_many",
+        on=["xbrl_factoid_calc"],
+        by=[p for p in pks if p != xbrl_factoid_name] + ["xbrl_factoid"],
+        value_col=params.column_to_check,
+    ).rename(columns={"xbrl_factoid": xbrl_factoid_name})
```

Comment:
this is an attempt to generalize calculating the calculated values but is currently NOT WORKING LOOK AWAY.

Reply:
it's working now 😎 ... now i'd like to use it over in output land.